src/main/java/com/aws/greengrass/deployment/DeviceConfiguration.java

```diff
@@ -226,6 +246,97 @@ private void initializeNucleusComponentConfig() {
+            DeploymentService.copyRecipeFileToComponentStore(componentStore, recipePath, logger);
+
+            // Update Nucleus in config store
+            Optional<ComponentRecipe> resolvedRecipe = kernel.getContext().get(RecipeLoader.class)
+                    .loadFromFile(new String(Files.readAllBytes(recipePath.toAbsolutePath()), StandardCharsets.UTF_8));
+            if (!resolvedRecipe.isPresent()) {
+                throw new PackageLoadingException("Failed to resolve Nucleus recipe");
+            }
+            ComponentRecipe componentRecipe = resolvedRecipe.get();
+            nucleusComponentName = componentRecipe.getComponentName();
+
+            // Add Nucleus version
+            Semver componentVersion = componentRecipe.getVersion();
+            kernel.getConfig().lookup(SERVICES_NAMESPACE_TOPIC, nucleusComponentName,
+                    VERSION_CONFIG_KEY).withValue(componentVersion.toString());
+            kernel.getConfig().lookup(SETENV_CONFIG_NAMESPACE, GGC_VERSION_ENV).withValue(componentVersion.toString());
+
+            // Copy unpacked artifacts to component store
+            Path unpackArtifactPath = kernel.getContext().get(NucleusPaths.class).unarchiveArtifactPath(
+                    new ComponentIdentifier(nucleusComponentName, componentVersion), "aws.greengrass.nucleus");
+            Utils.copyFolderRecursively(unpackDir, unpackArtifactPath, REPLACE_EXISTING, NOFOLLOW_LINKS,
+                    COPY_ATTRIBUTES);
+            Permissions.setArtifactPermission(unpackArtifactPath, FileSystemPermission.builder()
+                    .ownerRead(true).ownerExecute(true).groupRead(true).groupExecute(true)
+                    .otherRead(true).otherExecute(true).build());
+
+            KernelConfigResolver kernelConfigResolver = kernel.getContext().get(KernelConfigResolver.class);
+            // Add Nucleus dependencies
+            Map<String, DependencyProperties> nucleusDependencies = componentRecipe.getDependencies();
+            if (nucleusDependencies == null) {
+                nucleusDependencies = Collections.emptyMap();
+            }
+            kernel.getConfig().lookup(DEFAULT_VALUE_TIMESTAMP, SERVICES_NAMESPACE_TOPIC,
+                    nucleusComponentName, SERVICE_DEPENDENCIES_NAMESPACE_TOPIC)
+                    .dflt(kernelConfigResolver.generateServiceDependencies(nucleusDependencies));
+
+            // Add Nucleus lifecycle (after config interpolation)
+            Object interpolatedLifecycle = kernelConfigResolver.interpolate(componentRecipe.getLifecycle(),
+                    new ComponentIdentifier(componentRecipe.getComponentName(), componentRecipe.getVersion()),
+                    nucleusDependencies.keySet(),
+                    kernel.getConfig().lookupTopics(SERVICES_NAMESPACE_TOPIC).toPOJO());
+            nucleusLifecycle.replaceAndWait((Map<String, Object>) interpolatedLifecycle);
```

Comment:
Should this be `replaceAndWait()` or `dflt()` ? What happens when Nucleus restarts and already has lifecycle config from the tlog?

Reply:
It's prevented with this check https://github.com/aws-greengrass/aws-greengrass-nucleus/pull/806/files/77b497ecff63d8fdcc1771c88d586700ab512160#diff-3de627667101ba5b23e01542a659fd03c9a728449b99d59d0c309ccdbe5badc5R276-R281

Topics only has `updateFromMap` and `replaceAndWait`, but no `dflt`